### ISSUE TITLE
Authagraph projection?

### ISSUE THREAD

**Issue by jdu** [d3/d3-geo-projection#73]
Been using D3 for a while and came across this:

http://www.spoon-tamago.com/2016/10/28/hajime-narukawa-authagraph/

Less an issue, more of a wow check out this projection in case someone is interested in implementing it. I'm not even remotely enough of a mapping person to even know where to start implementing this, but it seems like it would be a pretty valuable type of projection for a lot of people (me included).

**Comment by Fil** [comment]
And now there is an (open source) pseudo-Authagraph, which is quite a little bit different from the (closed source) Authagraph.

https://marcinciura.wordpress.com/2016/11/20/authagraph/

**Comment by mciura** [comment]
Thanks for the mention, @Fil. FWIW, by moving Iceland to the opposite side of the map, I can achieve greater similarity to the original. More importantly, my current way of interpolating near the edges of the tetrahedron is imperfect: the first derivative of what comes out from the longitude is not continuous. You can see this effect in the graticule. I can find time to fix the issue if there is any interest in the code.

**Comment by espinielli** [comment]
@mciura very nice description and implementation!

**Comment by milimetric** [comment]
@mciura I've been thinking of using an authagraph projection in Wikimedia's new stats website, prototyped here: https://analytics-prototype.wmflabs.org/ (just a prototype, will be a few months before there's real data behind it).  On the detail page there's a map option for graphing some of the metrics, and we definitely want to offer either a Peters projection or something more accurate than Mercator (for data).  So your authagraph would be great, I'll take a closer look at it in a few weeks.

**Comment by mciura** [comment]
Feel free to ask me any questions if some day you translate my C code into whatever language you use, @milimetric

The discontinuity I mentioned above stems from using a home-made variation of quadratic Bézier curves in lines 485 and 505–515 of [my code](https://bitbucket.org/mciura/pseudo-authagraph/src/54f4d5080738a6cc6a4f9bb85742256b63518bb7/PJ_augr.c?at=default): the speed of a point moving along such a curve varies a lot when the parameter t changes linearly (and t is not even linear in my version).  Although we cannot use constant-speed Bézier curves, reading articles on them does not hurt.  In pseudo-Authagraph, the derivative of the longitude (i.e. the "speed" of the point) inside the interpolated region (i.e. minh < border, see line 480) should not vary a lot and it should continuously merge with the derivative of the longitude outside it (i.e. minh ≥ border).

**Comment by espinielli** [comment]
@mciura be wary of map projections: there is no such thing as an *accurate* map projection.
(also the [hype](https://www.theguardian.com/education/2017/mar/19/boston-public-schools-world-map-mercator-peters-projection) on Peters is much overdue according to tweets by professional carthographers [not me] I read)

See http://brilliantmaps.com/xkcd/ for a nice summary.

HTH

**Comment by jrus** [comment]
@milimetric please don’t use the Peters projection for any purpose. It causes obscene amounts of shape distortion. If you need an equal-area projection, be prepared to relax the condition that north is always up.

**Comment by milimetric** [comment]
I'm ok with relaxing that North is always up, but why does the Peters shape distortion matter?  What other projections do you recommend for equal-area?

**Comment by jrus** [comment]
If it must be equal area and show the world with no breaks, you could try one of:
https://en.wikipedia.org/wiki/Mollweide_projection
https://en.wikipedia.org/wiki/Eckert_IV_projection
https://en.wikipedia.org/wiki/Tobler_hyperelliptical_projection
https://en.wikipedia.org/wiki/Hammer_projection

The more breaks you’re willing to add the less distortion you’ll get. Likewise you’ll get much less shape distortion if you don’t require that the projection be equal area.

**Comment by ehj** [comment]
I know what I want for my future geography pupils :-)
An infinite plane of tessellated AuthaGraph-style tetrahedrons that you can 1) rotate as you like 2) zoom as you like and 3) as you like, choose the radius and centre of a circle on earth where there is no distortion. Any takers? :-)
I'd ask them to put Buffalo and Chicago on one circle and Stockholm and St Petersburg on another and make sure their radiuses are the same. Or Iceland and Lake Victoria, etc. 
//Erik

**Comment by Fil** [comment]
There is a very nice “AuthaPower” projection by @jkunimune15, explained at https://imgur.com/a/M0r5J
https://i.imgur.com/CZoQzOl.png
the code (in Java) is published under the MIT license at https://github.com/jkunimune15/Map-Projections

(That repo has in fact many other projections, including a few we don't have in D3.)

**Comment by toja** [comment]
I'm not the author of the post, it's probably @TJung1968

**Comment by jkunimune15** [comment]
@Fil Thanks for the mention! I made a better-formatted version of that imgur post on my wordpress here:

[https://kunimune.home.blog/2017/11/23/the-secrets-of-the-authagraph-revealed/](https://kunimune.home.blog/2017/11/23/the-secrets-of-the-authagraph-revealed/)

That one has the equations nicely formatted and stuff. If ye're interested in implementing it and have any questions, let me know.

**Comment by Fil** [comment]
oops sorry @toja and @TJung1968

Thank you @jkunimune15 for chiming in. I've ported your code to javascript/D3 [here](https://beta.observablehq.com/d/01a73895203dcdfb).

The equations work perfectly. However I still have to correctly follow all the final dance steps (rotate/translate/cut) to make it into the proper aspect ratio.

![image](https://user-images.githubusercontent.com/7001/52817924-99d32700-30a5-11e9-9207-d00aa89e1955.png)

**Comment by Fil** [comment]
It's getting closer

![image](https://user-images.githubusercontent.com/7001/52826263-6e116a80-30c0-11e9-970b-c4f608eb6f4c.png)

**Comment by TJung1968** [comment]
@Fil On your "settle on a good name" issue…
Since it’s very much but not exactly like the AuthGraph projection, I decided to call it "Kunimune's Mutated AuthaGraph Offspring" or "Kunimune's MAGO" for short on my website. @jkunimune15 seemed to like that name.

Just a suggestion, of course… Won’t bite you if you decide to choose another name. ;-)

Great work, thank you!

**Comment by ehj** [comment]
I'd go for LibreAuthaGraph :-) (or some permutation thereof)

But first, I want to say thanks! Fantastic!!

I now on paternity leave, but I can hardly wait to print a couple of 
tessellated flat earths on A3 and ask my class cut out and fold their 
favorite perspective into/onto a tetrahedron.

And I cannot hold back that I'd love to have a LibreAuthaGraph as a 
debian stable screensaver where a circle with varying diameter is 
bouncing across the screen where on the circle's perimeter, all relative 
distances are true*. Of course the map itself could move with time so 
that you always look at twelve o'clock etc etc :-)

Thanks again!

Best regards.

//Erik

*) I'm just thinking loud, but if you put a sphere that is almost as big 
as a tetrahedron inside a tetrahedron and project (from their common 
center) the sphere's surface onto the surface of the tetrahedron, the 
intersection of the bodies would be such circles.

Den 2019-02-15 kl. 17:54, skrev TJung1968:

**Comment by TJung1968** [comment]
Regarding "LibreAuthaGraph":
It has a ring to it, but… At the risk of sounding pedantic, I’d avoid using "AuthaGraph" in the name for two reasons:
1. The differences to the original AuthaGraph are, in my opinion, too big to use the same name;
2. I think the name was ill-chosen in the first place (by Mr. Narukawa): The "Autha" is derived from authalic = equal-area, but the projection isn’t equal-area.

Of course that is just my two cents worth.

**Comment by Fil** [comment]
The projection now fully works with its inverse. I have no clue how to select the correct rotations (especially for M2 and M3).

**Comment by Fil** [comment]
The Sphere is now correct, though it's a bit painful to draw it with a 100 tiny bits of projected great circles when it's in fact just a planar rectangle. But d3-geo currently doesn't have this type of planar clipping.

Re: the name, Infinitesimal Mutated AuthaGraph Offspring gives "*imago*" (as in *imago mundi*), maybe @jkunimune15 will like it.

**Comment by ehj** [comment]
+1 /imago mundi/

In that trajectory:


33 Varignon, Pierre. "1687. Projet d’une nouvelle méchanique avec un 
examen de l’opinion de M. Borelli sur les proriétez des poids suspendus 
par des cordes.".

//Erik
Den 2019-02-18 kl. 18:15, skrev Philippe Rivière:

**Comment by jkunimune15** [comment]
I am always in favour of backronyms!

**Comment by Fil** [comment]
For the rotation and translate I've made a superposition game:
https://beta.observablehq.com/d/01a73895203dcdfb#r

Best fit so far `r = [18.05, -12.75, 2.8, 1.16]` but ymmv
![authafit](https://user-images.githubusercontent.com/7001/53268726-2fdf0100-36e7-11e9-97f2-e8fd87c90855.png)

(Note that are three additional parameters: scale and translate)

**Comment by Fil** [comment]
Playing again over the week-end with this projection, I find it easier to fit with the original design by using _k = 0.68_ rather than _k=0.7_.

I would go further: in my view, the map is more comfortable with a much lower exponent, like _k=0.62_ or 0.6: I don't know how much that costs in terms of distortion, but with a lower _k_ you don't get the feeling that Brazil is "wrong".

See
https://beta.observablehq.com/d/d9209b3ee923f4b4

(I will push the code to a branch of d3-geo-polygon in a few hours.)

**Comment by waldyrious** [comment]
> I would go further: in my view, the map is more comfortable with a much lower exponent, like _k=0.62_ or 0.6: I don't know how much that costs in terms of distortion, but with a lower _k_ you don't get the feeling that Brazil is "wrong".

From a purely aesthetic perspective, I gotta say that going even lower makes the map "feel" significantly less distorted in the landforms (whether this is actually true is another topic, but it could be interesting to measure distortion restricted only to the land areas).

Here's with k = 0.55:

![screenshot_2019-02-24 the imago projection](https://user-images.githubusercontent.com/478237/53307249-673be200-388e-11e9-8cf0-c3b0d526f7e9.png)

Besides Brazil, it seems to me that Africa, Australia, Asia, Siberia and Greenland all look closer to their actual shapes.

**Comment by mciura** [comment]
If it helps, I can share my Python program with:
- a list of 461 land points on a more or less evenly spaced mesh, precomputed from Natural Earth data,
- code to compute the total angular distortion in these points.
Translating it to JS should be easy.

**Comment by ehj** [comment]
Apologies for my ignorance, but is it possible to tessellate the imago 
mundi map regardless of the values of k and shift?

To be able to fold any cutout from a A3 print of an imago mundi map onto 
a tetrahedron is what I have in mind, like this:

https://img.newatlas.com/authograph-world-map-projection-6.jpg?auto=format%2Ccompress&fit=max&q=60&w=1000&s=3f9cfbc9bc806ed9941c34fd8ab2b96d

Thanks!

//Erik

**Comment by Fil** [comment]
Hi @ehj, 

It should work - the periodicity of the tiles is independent of the way we arrange the contents on each tile.

(Note that it's not "any cutout" that will fold properly, but only cutouts at specific angles and arranged along certain lines, as in the schema.)

It would be great to have a notebook to demonstrate this property.

I've started something here but it's still missing a few things, like the bold rectangle on the right, the triangle, the tile pattern etc.
https://beta.observablehq.com/d/5ffda0a0690aa02f

Help appreciated! It's also a great way to test the projection (it looks like there is a bug with the Sphere in some cases…).

**Comment by Fil** [comment]
@mciura I was indeed working on a notebook for the distorsion, which you can check at
https://beta.observablehq.com/d/8941e2624ed51f6d
(not finished… I'm not sure whether trying to cap the maximum is the best strategy)

From my tests it appears that the optimal value (to minimize angular distorsion on land) is for _k_ = 0.59.

Again, don't hesitate to play with it and suggest changes!

**Comment by TJung1968** [comment]
Thanks a lot @Fil – your tool is really helpful in evaluating the distortion patterns!

First off, I’d like to express my hope that _k_ (and _shift_ as well) will not be hard coded but can be set by the user as parameters. (I guess you did have this in mind but no one voiced it so far…)
But that of course leaves the decision what the default value will be. And here, I’m a bit torn:

1. On one hand, I tend to assume that a lot of users will hope for a projection that is as close to the original AuthaGraph as possible, which suggest to choose _k = 0.68_ as default.
2. On the other hand, I agree that values around _0.6_ look better. Moreover, since the AuthaGraph isn’t authalic anyway, I think it doesn’t hurt to take a few further steps away from equivalence in order to obtain better shapes.

In other words: I really don’t know what’s best. :-/

**Comment by Fil** [comment]
My thinking is that the "optimal" k=0.59 would mark the true personality of Imago, whereas k=0.68 would only ever be an approximation of another projection, a shadow.

**Comment by TJung1968** [comment]
Very good point!

**Comment by Fil** [comment]
The imago branch will be merged from https://github.com/d3/d3-geo-polygon/pull/24
Please test and comment there.

**Comment by Fil** [comment]
The Imago projection has been released with d3-geo-polygon 1.7.0 !

**Comment by Droyk** [comment]
Possible to add this map in the interactive version i.e. project transition https://bl.ocks.org/mbostock/3711652
